### PR TITLE
Stat filters 2

### PR DESCRIPTION
**Pull request by sundevour**
`maxstatloadout` for finding an equippable set that gets you the highest X stat. a mini loadout builder.

`maxstatvalue` for finding the top values for each class+slot combination
if my best intellect titan helmet is 10, this will highlight all int 10 titan helmets

next up: `maxbasestatvalue`

can i start hyphenating these? they are getting unmanageable

### CODE REVIEW

**Comment by bhollis** [comment]
I have an idea in the back of my mind that stops requiring us to invent a complicated language for these in order to use them, but I haven't managed to get to work on it.

File: config/i18n.json [hunk 1]
```diff
@@ -389,8 +389,8 @@
   "ItemMove": {
     "Consolidate": "Consolidated {{name}}",
     "Distributed": "Distributed {{name}}\n {{name}} is now equally divided between characters.",
-    "MovingItem_male": "Transfer to {{target}}",
-    "MovingItem_female": "Transfer to {{target}}",
+    "MovingItem_male": "Transferring to {{target}}",
+    "MovingItem_female": "Transferring to {{target}}",
```
Comment (bhollis) [inline config/i18n.json]: This seems unrelated. I had it as "Transfer" since it'll also show this message for a bit after the transfer is done - the progress notification is about the transfer itself.

File: src/app/search/search-filters.ts [hunk 2]
```diff
@@ -437,6 +443,35 @@ function searchFilters(
...
+                // just assign if this is the first
+                !(stat.statHash in _maxStatValues[itemSlot])
+                  ? stat.value
+                  : // else we are looking for the biggest stat
+                    Math.max(_maxStatValues[itemSlot][stat.statHash], stat.value);
+            }
+          }
+        }
+      }
+      console.log(_maxStatValues);
```
Comment (bhollis) [inline src/app/search/search-filters.ts]: Remove?

File: config/i18n.json (hunk 1 above)
Comment (sundevour) [reply config/i18n.json]: oh dim.json in master says "transferring" so i was just updating root string to match. undone.

**Comment by sundevour** [comment]
docs updated
